test/dart3_suggestors/null_safety_prep/required_flux_props_test.dart

```diff
@@ -0,0 +1,1160 @@
+              });
+
+              test('and the types match:', () async {
+                await testSuggestor(
+                  expectedPatchCount: 2,
+                  input: withFluxComponentUsage('''
+                    // ignore: undefined_identifier
+                    UiFactory<FooConsumerProps> FooConsumer = castUiFactory(_\$FooConsumer);
+                    class FooConsumerProps = UiProps with FluxUiPropsMixin<FooActions, FooStore>;
+                    class FooConsumerComponent extends FluxUiComponent2<FooConsumerProps> {
+                      someMethod() {
+                        return ${maybeInvokeBuilder('''Foo()
+                          ..id = '123'
+                        ''')}
+                      }
+      
+                      @override
+                      render() => null;
+                    }
+                  '''),
+                  expectedOutput: withFluxComponentUsage('''
+                    // ignore: undefined_identifier
+                    UiFactory<FooConsumerProps> FooConsumer = castUiFactory(_\$FooConsumer);
+                    class FooConsumerProps = UiProps with FluxUiPropsMixin<FooActions, FooStore>;
+                    class FooConsumerComponent extends FluxUiComponent2<FooConsumerProps> {
+                      someMethod() {
+                        return ${maybeInvokeBuilder('''Foo()
+                          ..store = props.store
+                          ..actions = props.actions
+                          ..id = '123'
+                        ''')}
+                      }
+      
+                      @override
+                      render() => null;
+                    }
+                  '''),
+                );
+              });
+            });
+          });
+        });
+      });
+    }
+
+    sharedTests(invokeBuilder: false);
+    sharedTests(invokeBuilder: true);
+  });
+}
+
+String withFluxComponentUsage(String source,
+    {String? actionsName = 'FooActions', String? storeName = 'FooStore'}) {
+  String getActionsClasses() => actionsName == null
+      ? ''
+      : '''
+class $actionsName {
+  $actionsName();
+}
+
+class Baz$actionsName {
+  Baz$actionsName();
+}
+''';
+
+  String getStoreClasses() => storeName == null
+      ? ''
+      : '''
+class $storeName {
+  $storeName();
+}
+
+class Baz$storeName {
+  Baz$storeName();
+}
+''';
+
+  return withOverReactImport('''$source
+
+${getActionsClasses()}
+
+${getStoreClasses()}
+
+UiFactory<FooProps> Foo = castUiFactory(_\$Foo); // ignore: undefined_identifier
+
+class FooProps = UiProps with FluxUiPropsMixin<${actionsName ?? 'Null'}, ${storeName ?? 'Null'}>;
+
+class FooComponent extends FluxUiComponent2<FooProps> {
+  @override
+  render() => null;
+}
+
+UiFactory<NotFooProps> NotFoo = castUiFactory(_\$NotFoo); // ignore: undefined_identifier
+
+// FIXME (adl): Do we need to have a mixin that has store/actions with the correct type? (See FIXME in impl regarding out-of-scope vars being used) 
```

Comment:
Should this fixme be addressed?

Reply:
It already has been (via class / function props consumption tests)